### DISCUSSION TITLE
is `vue-component-meta` only designed for vue ts files

### DISCUSSION THREAD

**Discussion by damvaz** [vuejs/language-tools#5042] [Q&A]
Hi!! I'm new in vue language tools, sorry if i make a basic question :-) 

Is `vue-component-meta` only designed to detect metadata from SFC with `lang=ts` attribute? or it can also detect plain vue js files?

For example:
```html
<script>
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'MyAwesomeComponent',
  props: {
    /**
    Some description
   */
    aProp: {
      type: String,
      default: 'Hello!'
   }
  },
  setup() {
    // some code
  }
})
</script>

<template>
  <div>My Awesome Component</div>
</template>
```

Thanks for any answer 🤍

**Comment by KazariEX** [comment]
If you use the same API (script setup or options), it generates almost the same virtual code for `lang="js"` or `lang="ts"`, so I think it should work well in your use case.

**Reply by damvaz** [reply]
Thanks @KazariEX
